service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterMapper.java

```diff
@@ -64,7 +64,7 @@ public String mapEncounterToEhrComposition(Encounter encounter) {
             .originalText(buildOriginalText(encounter));
 
         final String recReference = findParticipantWithCoding(encounter, ParticipantCoding.RECORDER)
-            .map(idMapper::get)
+            .map(messageContext.getAgentDirectory()::getAgentId)
```

Comment:
Could consider extracting `messageContext.getAgentDirectory()` to a variable, since it's called twice in this method.

It can be important to note that if the thing on the left of the `::` in a method reference needs evaluating, it is evaluated regardless of whether or not the lambda is actually invoked. In this case that means that it really is a refactoring and we're not adding any calls that wouldn't be made anyway.